google/cloud/forseti/services/inventory/storage.py

```diff
@@ -154,6 +155,81 @@ def set_error(self, session, message):
         session.add(self)
         session.flush()
 
+    def get_lifecycle_state_details(self, session, resource_type_input):
+        """Count of lifecycle states of the specified resources.
+
+        Generate/return the count of lifecycle states (ACTIVE, DELETE_PENDING)
+        of the specific resource type input (project, folder) for this inventory
+        index.
+
+        Args:
+            session (object) : session object to work on.
+            resource_type_input (str) : resource type to get lifecycle states.
+
+        Returns:
+            dict: a (lifecycle state -> count) dictionary
+        """
+        resource_data = Inventory.resource_data
+
+        details = dict(
+            session.query(func.json_extract(resource_data, '$.lifecycleState'),
+                          func.count())
+            .filter(Inventory.inventory_index_id == self.id)
+            .filter(Inventory.category == 'resource')
+            .filter(Inventory.resource_type == resource_type_input)
+            .group_by(func.json_extract(resource_data, '$.lifecycleState'))
+            .all())
+
+        for key in details.keys():
+            new_key = key.replace('\"', '').replace('_', ' ')
+            new_key = ' - '.join([resource_type_input, new_key])
+            details[new_key] = details.pop(key)
+
+        if len(details) == 1:
```

Comment:
if len is 0, do we still want 2 entries with count = 0?

Reply:
If len is 0, then that resource_type does not exist - would not show up in the Inventory Summary part either. Might be confusing for the user to see a resource_type in the Details section but not in the Summary section. This is a good outlier case - if you think it should be included, let me know!